City-of-Helsinki/drupal-hdbt
# UHF-9156: Update HDS to v3

# [UHF-9156](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9156)


## What was done


As fas as I know HDS package changes should only affect our React searches. HDS styles we use are manually copied to HDBT and locked to certain version because of that.

* Update HDS packages to version 3 and HDS dependencies
* Stop using HDS Tabs in School search. Instead use our own code
* Fix use of html-parser in Ploughing search
* Update other React/js packages that had dependapot PR:s open
* Update node version to 18.19.0
* Remove Joi, es6-promise, Buffer, path-browserify packages because those seemed unused
* Add `npm audit fix` fixes

## How to install

* Make sure these instance is up and running on latest dev branch:
* Etusivu, Rekry, KYMP, KASKO, SOTE
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-9156-update-hds-to-v3`
* Run `make shell`
* Run `drush sapi-rt; drush sapi-c; drush sapi-i; drush cr`

## How to test


* [x] Test all react searches and check that they work and look good
* [x] Rekry: Test job search: https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/etsi-avoimia-tyopaikkoja
* [x] KASKO: Test school search: https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/perusopetus/peruskoulut
* [x] Etusivu: https://helfi-etusivu.docker.so/fi/uutiset
* [x] SOTE: Add health station paragraph to Landing page (Health station React search is not published yet)
* [x] SOTE: Add Maternity and child health clinic paragraph to Landing page (Maternity and child health clinic React search is not published yet)
* [x] SOTE (or some other instance is fine too :)): test event search: https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/palvelukeskukset & https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/palvelukeskukset/kannelmaen-palvelukeskus
* [x] KYMP: https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/kaupunkisuunnittelu-ja-rakentaminen/hae-suunnitelmia-ja-hankkeita
* [x] KYMP: Add Ploughing search (aurausaikatauluhaku) to basic page. (Ploughing search is not published yet)

Check also running the NPM commands with some Drupal instance.
* [x] Install HDBT: `composer reinstall drupal/helfi_platform_config --prefer-source`
* [x] Go to the root of HDBT theme
* [x] Install it: `nvm use; npm i;`
* [x] Check that running npm "scripts" from package.json file work. You may need to add some code changes to check all the command e.g. linting.



[UHF-9156]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
